GitUI/Editor/FileViewer.cs

```diff
@@ -1756,116 +1699,32 @@ private void CopyPatchToolStripMenuItemClick(object sender, EventArgs e)
 
         private void copyNewVersionToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            CopyNotStartingWith('-');
+            internalFileViewer.CopyNotStartingWith('-');
         }
 
         private void copyOldVersionToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            CopyNotStartingWith('+');
+            internalFileViewer.CopyNotStartingWith('+');
         }
 
         /// <summary>
         /// Go to next change
-        /// For normal diffs, this is the next diff
+        /// For normal diffs, this is the next block of lines with a difference.
         /// For range-diff, it is the next commit summary header.
         /// </summary>
-        /// <param name="sender">sender object.</param>
-        /// <param name="e">event args.</param>
         private void NextChangeButtonClick(object sender, EventArgs e)
         {
             Focus();
 
-            string[] inChange = _viewMode == ViewMode.RangeDiff
-                ? _rangeDiffPrefixes
-                : DiffHighlightService.IsCombinedDiff(internalFileViewer.GetText())
-                    ? _combinedDiffPrefixes
-                    : _normalDiffPrefixes;
-
-            // skip the first pseudo-change containing the file names for normal diffs
-            int headerEndLine = _viewMode == ViewMode.RangeDiff ? 0 : 4;
-            int currentVisibleLine = internalFileViewer.LineAtCaret;
-            int startLine = Math.Max(headerEndLine, currentVisibleLine + 1);
-            int totalNumberOfLines = internalFileViewer.TotalNumberOfLines;
-            bool emptyLineCheck = _viewMode == ViewMode.RangeDiff;
-            for (int line = startLine; line < totalNumberOfLines; line++)
-            {
-                string lineContent = internalFileViewer.GetLineText(line);
-
-                if (_viewMode == ViewMode.RangeDiff ^ lineContent.StartsWithAny(inChange))
-                {
-                    if (emptyLineCheck)
-                    {
-                        internalFileViewer.FirstVisibleLine = Math.Max(line - headerEndLine, 0);
-                        internalFileViewer.LineAtCaret = line;
-                        return;
-                    }
-                }
-                else
-                {
-                    emptyLineCheck = true;
-                }
-            }
-
-            // Do not go to the end of the file if no change is found
-            ////TextEditor.ActiveTextAreaControl.TextArea.TextView.FirstVisibleLine = totalNumberOfLines - TextEditor.ActiveTextAreaControl.TextArea.TextView.VisibleLineCount;
+            internalFileViewer.GoToNextChange(NumberOfContextLines);
         }
 
         private void PreviousChangeButtonClick(object sender, EventArgs e)
         {
             Focus();
 
```

Comment:
```suggestion
```

Reply:
Have a look in the function called at the next line. 😉
Or didn't you intend to move this, too?

Reply:
Did not intend to change that in FileViewer.cs, removing in FileViewerInternal.cs